### PR TITLE
Division 293 and concessional contributions variables

### PR DESCRIPTION
**Pull request by HughParsonage**

### CODE REVIEW

**Comment by HughParsonage** [comment]
These functions require testing.

File: R/mutate_super_vars.R [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+             "MCS_Prsnl_Contr", 
+             "MCS_Othr_Contr", 
+             "MCS_Ttl_Acnt_Bal") %in% names(.sample.file))){
+    stop("The sample file you requested does not have the variables needed for this function.")
+  }
+  
+  # Concessional contributions
+  if (use_other_contr){
+    .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt + MCS_Othr_Contr]
+  } else {
```
Comment (youncg81) [inline R/mutate_super_vars.R]: I've changed my mind on this.  We should not add any of the other contributions to concessional contributions.  My concern is that none of the items are actually concessional contributions, they are either after-tax contributions of DB fund notional contributions, neither of which are taxed are per concessional contributions.

We could consider adding it to notional concessional contributions for div 293 purposes only.

File: R/mutate_super_vars.R [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  .sample.file[ , low_tax_contributions_div293 := pmaxC(concessional_contributions - excess_concessional_contributions, 0)]
+  .sample.file[ , div293_income := surchargeable_income_div293 + low_tax_contributions_div293]
+  .sample.file[ , div293_tax := ifelse(div293_income > div293_threshold, 
+                                       0.15 * pminV(low_tax_contributions_div293, 
+                                                    pmaxC(div293_income - div293_threshold, 0)), 
+                                       0)]
+  
+  # Modify taxable income to reflect exceeding cap:
+  # Note dot (too much to ask that the column wouldn't already exist)
+  .sample.file[ , .new_Taxable_Income := Taxable_Income + excess_concessional_contributions]
```
Comment (youncg81) [inline R/mutate_super_vars.R]: Lets change the variable name here.  New taxable income is a bit generic, how about:

Taxable_income_for_ect

File: R/mutate_super_vars.R [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+             "MCS_Ttl_Acnt_Bal") %in% names(.sample.file))){
+    stop("The sample file you requested does not have the variables needed for this function.")
+  }
+  
+  # Concessional contributions
+  if (use_other_contr){
+    .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt + MCS_Othr_Contr]
+  } else {
+    .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt]
+  }
```
Comment (youncg81) [inline R/mutate_super_vars.R]: we should add something like the below to split contributions by type:

sg_conts = max ( MCS_Emplr_Contr - Rptbl_Empr_spr_cont_amt , 0) 
salary_sacrifice_conts = Rptbl_Empr_spr_cont_amt
personal_deductible_conts = Non_emp_spr_amt

File: R/mutate_super_vars.R [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+  new_sample_file[, new_revenue := income_tax(new_Taxable_Income, fy.year) + new_div293_tax]
+  data.table::setkey(new_sample_file)
+  sample_file[new_sample_file]
+}
+
+n_affected_from_cap_and_div293 <- function(...){
+  revenue_from_cap_and_div293(...) %$%
+    {sum(abs(.[["prv_revenue"]] - .[["new_revenue"]]) > 1) * .[["WEIGHT"]][1]}
+}
+
```
Comment (youncg81) [inline R/mutate_super_vars.R]: The rest looks good to me.

File: R/mutate_super_vars.R (hunk 3 above)
Comment (HughParsonage) [inline R/mutate_super_vars.R]: Patched as commit 8a2b4038f30f207a741f037468a12fd440c0788a

File: R/mutate_super_vars.R (hunk 2 above)
Comment (HughParsonage) [inline R/mutate_super_vars.R]: What is `ECT`?

Comment (HughParsonage) [inline R/mutate_super_vars.R]: Patched as a84cbba0f440828af01f04cf7bc0264078ff8911

File: R/mutate_super_vars.R (hunk 1 above)
Comment (HughParsonage) [inline R/mutate_super_vars.R]: Patched as 52c8c58470417c747dee50fe20baec4b933b1b54

Comment (HughParsonage) [inline R/mutate_super_vars.R]: Note that by default this is not included in either. It's only if the user specifically requests it.

File: R/mutate_super_vars.R [hunk 5]
```diff
@@ -77,12 +82,13 @@ apply_super_caps_and_div293 <- function(.sample.file,
...
-  if (use_other_contr){
-    .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt + MCS_Othr_Contr]
-  } else {
-    .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt]
-  }
+  .sample.file[ , concessional_contributions := MCS_Emplr_Contr + Non_emp_spr_amt]
+  .sample.file[ , non_concessional_contributions := MCS_Prsnl_Contr - Non_emp_spr_amt]
+  
   if (age_based_cap){
     if (!any("age_range_description" == names(.sample.file))){
```
Comment (youncg81) [inline R/mutate_super_vars.R]: Just max off the non_conc_conts to zero

**Comment by youncg81** [comment]
Happy subject to addressing last comment

File: R/mutate_super_vars.R (hunk 5 above)
Comment (HughParsonage) [inline R/mutate_super_vars.R]: Patched in a171bef4787aa760bd8fb1c8cb6b83c61307e6a7

**Comment by HughParsonage** [comment]
Closes #14 .

**Pull request merged**